### PR TITLE
Limit jobs by filtering paths

### PR DESCRIPTION
**Pull request by bdauria**

### CODE REVIEW

**Review by Thebarda — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]
me 🤝 me

**Comment by bdauria** [comment]
image

File: .github/workflows/web.yml [hunk 1]
```diff
@@ -22,6 +22,38 @@ env:
...
+              - '**/*.[jt]sx?'
+              - '**/www/front_src/**'
+              - '**/tests/e2e/**'
+              - '**/package*'
+              - '**/lighthouse/**'
+              - '**/tsconfig.json'
+              - '**/cypress/**|**/pnpm-*'
+            has_backend_changes:
+              - '**/*.php'
+              - '**/phpstan.neon'
```
Comment (kduret) [inline .github/workflows/web.yml:45]:
```suggestion
              - '**/phpstan*.neon'
```

File: .github/workflows/web.yml [hunk 2]
```diff
@@ -22,6 +22,38 @@ env:
...
+              - '**/*.xml|**/bin/**'
+              - '**/tmpl/**'
+              - '**/features/**'
+              - '**/www/src/**'
+              - '**/config/**'
+              - '**/composer.*'
+              - '**/tests/api/**'
+              - '**/tests/php/**'
+              - '**/tests/clapi_export/**'
+              - '**/www/!(front_src)/**'
```
Comment (kduret) [inline .github/workflows/web.yml:55]:
```suggestion
              - '**/www/!(front_src)/**'
              - '**/doc/API/**
```

File: .github/workflows/web.yml [hunk 3]
```diff
@@ -22,6 +22,38 @@ env:
...
+              - '**/lighthouse/**'
+              - '**/tsconfig.json'
+              - '**/cypress/**|**/pnpm-*'
+            has_backend_changes:
+              - '**/*.php'
+              - '**/phpstan.neon'
+              - '**/*.xml|**/bin/**'
+              - '**/tmpl/**'
+              - '**/features/**'
+              - '**/www/src/**'
```
Comment (kduret) [inline .github/workflows/web.yml:49]:
```suggestion
              - '**/src/**'
```

Comment (bdauria) [reply .github/workflows/web.yml:49]: this will be picked up as a frontend change in @centreon/ui

Comment (kduret) [reply .github/workflows/web.yml:49]: Indeed, I've updated the path to be strict

Comment (Thebarda) [reply .github/workflows/web.yml:49]: This will take every file changed in `centreon/www/front_src/src`.
We may find another way for this one

**Review by kduret — APPROVED** [review]

**Review by paloth — APPROVED** [review]

**Pull request merged**